### PR TITLE
chore: linecol to bytepos

### PR DESCRIPTION
**Pull request by brandonspark**
## What:
This PR refactors the `bytepos_to_linecol_fun` type into a `pos_info` type, which contains both a `bytepos_to_linecol_fun` and a `linecol_to_bytepos_fun`.

## Why:
I needed the linecol to bytepos function, and it seemed easiest to piggyback off of the `bytepos_to_linecol_fun` logic, and compute both at the same time.

## How:
We use the previous PR https://github.com/semgrep/semgrep/pull/9222 to quickly search for the proper line/col, by binary searching over all the byte positions previously computed (and their associated line/cols).

It currently raises an exception on being given an invalid line/col.

## Test plan:
Automated tests

### CODE REVIEW

File: libs/lib_parsing/Pos.ml [hunk 1]
```diff
@@ -90,7 +90,10 @@ let string_of_pos { file; line; column; _ } = spf "%s:%d:%d" file line column
 (*****************************************************************************)
 
 (* conversion table, in the shape of a function *)
-type bytepos_to_linecol_fun = int -> int * int
+type pos_info = {
```
Comment (aryx) [inline libs/lib_parsing/Pos.ml]: maybe type bytepos_linecol_converters ?

File: libs/lib_parsing/Pos.ml [hunk 2]
```diff
@@ -106,32 +109,62 @@ type bytepos_to_linecol_fun = int -> int * int
  *   - in each lexer you need to take care of newlines and update manually
  *     the field.
  *)
-let complete_position filename table (x : t) =
+let complete_position filename pos_info (x : t) =
```
Comment (aryx) [inline libs/lib_parsing/Pos.ml]: maybe converters instead of pos_info ?

File: libs/lib_parsing/Pos.mli [hunk 3]
```diff
@@ -36,24 +36,30 @@ val string_of_pos : t -> string
...
 (*
    Return (line, column) from a byte position.
+   Also return byte position from a (line, column).
 
    If the byte position is out of range, the functions of this type return
    the nearest valid position which is either the first or the last position
    in the range.
    Empty files admit at least one valid byte position.
+
+   If the (line, column) is out of range, an exception will be raised.
```
Comment (aryx) [inline libs/lib_parsing/Pos.mli]: which exception?

**Review by aryx — CHANGES_REQUESTED** [review]

File: languages/yaml/generic/yaml_to_generic.ml [hunk 4]
```diff
@@ -593,7 +593,9 @@ let mask_unicode str =
 let parse_yaml_file ~is_target file str =
   (* we do not preprocess the yaml here; ellipsis should be transformed
    * only in the pattern *)
-  let charpos_to_pos = Some (Pos.full_charpos_to_pos_large file) in
+  let charpos_to_pos =
```
Comment (aryx) [inline languages/yaml/generic/yaml_to_generic.ml]: rename charpos to bytepos here (to be consistent).

Comment (aryx) [reply languages/yaml/generic/yaml_to_generic.ml]: and also because we operate at the bytepos level, not (unicode) character level.

**Review by aryx — APPROVED** [review]

**Comment by kopecs** [comment]
Formatting?

**Pull request merged**